File: gin/gin-chat/common/utils/writelog.go
package utils

import (
	"fmt"
	"gin_proj/config"
	"log"
	"os"
	"path/filepath"
	"time"
)

// writeLog 直接写入日志到文件
func WriteLog(message string, level string) {
	// 获取日志路径配置
	logPath := config.Config.Logs.Path

	if logPath == "" {
		// 如果没有配置日志路径，使用默认路径
		logPath = filepath.Join("logs")
	}
	// 如果日志路径不是绝对路径，则转换为绝对路径
	if !filepath.IsAbs(logPath) {
		// 获取当前工作目录
		wd, err := os.Getwd()
		if err != nil {
			panic(fmt.Errorf("获取当前工作目录失败: %s", err))
		}
		// 拼接成绝对路径
		logPath = filepath.Join(wd, logPath)
	}

	// 确保日志目录存在
	if err := os.MkdirAll(logPath, 0755); err != nil {
		panic(fmt.Errorf("创建日志目录失败: %s", err))
	}

	// 日志文件名 (按日期)
	today := time.Now().Format("2006-01-02")
	logFile := filepath.Join(logPath, fmt.Sprintf("app_%s.log", today))

	// 格式化日志消息
	timestamp := time.Now().Format("2006-01-02 15:04:05")
	logEntry := fmt.Sprintf("%s - %s - %s\n", timestamp, level, message)

	// 写入文件
	file, err := os.OpenFile(logFile, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
	if err != nil {
		panic(fmt.Errorf("打开日志文件失败: %s", err))
	}
	defer file.Close()

	if _, err := file.WriteString(logEntry); err != nil {
		panic(fmt.Errorf("写入日志文件失败: %s", err))
	}
}

// 便捷函数
func LogInfo(message string) {
	log.Println(message)
	WriteLog(message, "INFO")
}

func LogError(message string) {
	log.Println(message)
	WriteLog(message, "ERROR")
}

func LogWarning(message string) {
	log.Println(message)
	WriteLog(message, "WARNING")
}

func LogDebug(message string) {
	log.Println(message)
	WriteLog(message, "DEBUG")
}

// SimpleLogger 结构体，模仿Python的类
type SimpleLogger struct{}

func (l *SimpleLogger) Info(message string) {
	LogInfo(message)
}

func (l *SimpleLogger) Error(message string) {
	LogError(message)
}

func (l *SimpleLogger) Warning(message string) {
	LogWarning(message)
}

func (l *SimpleLogger) Debug(message string) {
	LogDebug(message)
}

// 创建全局logger实例
var FileLogger = &SimpleLogger{}

File: fastapi/fastapi-ai/common/utils/writeLog.py
import os
from datetime import datetime
from typing import Any


def _get_logger():
    # lazy import to avoid circular import with package-level imports
    try:
        from .loggers import logger
        return logger
    except Exception:
        import logging
        return logging.getLogger('app')


def write_log(message: str, level: str = "INFO") -> None:
    """
    直接写入日志到文件
    
    Args:
        message: 日志消息
        level: 日志级别 (INFO, WARNING, ERROR, DEBUG)
    """
    # lazy import load_config to avoid import-time circular dependency with config/nacos
    from config import load_config
    LOG_PATH: str = load_config("logs")["path"]
    if not os.path.exists(LOG_PATH):
        os.makedirs(LOG_PATH, exist_ok=True)
    log_file: str = os.path.join(LOG_PATH, f"app_{datetime.now().strftime('%Y-%m-%d')}.log")
    timestamp: str = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
    log_entry: str = f"{timestamp} - {level} - {message}\n"
    with open(log_file, 'a', encoding='utf-8') as f:
        f.write(log_entry)

def log_info(message: str) -> None:
    _get_logger().info(message)
    write_log(message, "INFO")

def log_error(message: str) -> None:
    _get_logger().error(message)
    write_log(message, "ERROR")

def log_warning(message: str) -> None:
    _get_logger().warning(message)
    write_log(message, "WARNING")

def log_debug(message: str) -> None:
    _get_logger().debug(message)
    write_log(message, "DEBUG")

class SimpleLogger:
    def info(self, message: str) -> None:
        log_info(message)
    
    def error(self, message: str) -> None:
        log_error(message)
    
    def warning(self, message: str) -> None:
        log_warning(message)
    
    def debug(self, message: str) -> None:
        log_debug(message)

fileLogger: SimpleLogger = SimpleLogger()
